new-projects.js

```diff
@@ -0,0 +1,5 @@
+exports.getNewProjects = (currentProjects, targetProjects) => {
+  const currentProjectNames = currentProjects.map((project) => project.node.project.name);
```

Comment:
The names of the arguments (and also the GitHub Action inputs) don't indicate that `currentProjects` is a list of nodes of project, while `targetProjects` is just a list of project names. I'd recommend to adjust the naming, e.g. `targetProjectNames`.